Feature | Ausbau Stufen Gradle Plugin

ExLeonem opened the pull request:
### Description
- update documentation
  - switch from `projectStage` to `status`
  - add migration guide
- update config file to use new status

### How to test

#### Old plugin tasks still working

1. Try the following tasks with the old plugin version
   - `uploadFormularFiles`
   - `uploadProcessModelFiles`
   - Tasks should still work when using `projectStage`

#### Switch from `projectStage` to `status` working

1. Switch to newest Plugin Version
2. Execute tasks
   - `uploadFormularFiles`
   - `uploadProcessModelFiles`
   - Tasks should work with the new `status` attribute in default.config


### Deployment-Locations

Check other PR's in bitbucket for deployment locations

Commit:
update docs

SBW-25815

- replace information about "stufe"
with information about "status"
- add migration guide and deprecation
warning to the documentation
- update the config file
- add breaking changes file

Commit:
Minor changes

SBW-25815

- revert url in default.json
- change attribute name in docs

mattbond commented on migration.md:
```diff
@@ -0,0 +1,56 @@
+Die Schnittstelle selbst funktioniert übergangsweise sowohl mit der Stufe, als auch dem neuen Status.
+Es darf jedoch nur einer der beiden Werte an die Schnittstelle übertragen werden.
+
+**Wichtig:** Ab einem einem Zeitpunkt X, wird die Verwendung der Projektstufen vollständig ausgebaut 
```
Perhaps use wording like "deprecated"

mattbond commented on migration.md:
```diff
@@ -0,0 +1,56 @@
+| FUNCTIONAL_ANALYSIS      | EDIT   |
+| TECHNICAL_IMPLEMENTATION | EDIT   |
+| QUALITY_ASSURANCE        | TEST   |
+| CERTIFIED                | FINAl  |
```
Typo FINAl  -> FINAL

Commit:
Incorporate QS corrections

SBW-25815

- add example
- fix mapping table
- switch phrasing from "wichtig" to "deprecated"
- eliminate duplicate sentence
- fix link

ExLeonem commented:
Erst Mergen wenn die neuste Version des Plugins selbst ausgeliefert wurde.

Commit:
Add link to migration guides to readme

SBW-25815

Commit:
Change wording in schnittstelle.md

SBW-25815

- change wording from "achtung" to "deprecated"

mattbond commented:
The version number of the plug-in in build.gradle needs to be updated one published.
